src/auto/tree_model_sort.rs

```diff
@@ -0,0 +1,96 @@
+// This file was generated by gir (https://github.com/gtk-rs/gir)
+// from gir-files (https://github.com/gtk-rs/gir-files)
+// DO NOT EDIT
+
+use TreeDragSource;
+use TreeIter;
+use TreeModel;
+use TreePath;
+use TreeSortable;
+use ffi;
+use glib::object::IsA;
+use glib::translate::*;
+use glib_ffi;
+use gobject_ffi;
+use std::mem;
+use std::ptr;
+
+glib_wrapper! {
+    pub struct TreeModelSort(Object<ffi::GtkTreeModelSort, ffi::GtkTreeModelSortClass>): TreeDragSource, TreeModel, TreeSortable;
+
+    match fn {
+        get_type => || ffi::gtk_tree_model_sort_get_type(),
+    }
+}
+
+pub trait TreeModelSortExt {
+    fn clear_cache(&self);
+
+    fn convert_child_iter_to_iter(&self, child_iter: &TreeIter) -> Option<TreeIter>;
+
+    fn convert_child_path_to_path(&self, child_path: &TreePath) -> Option<TreePath>;
+
+    fn convert_iter_to_child_iter(&self, sorted_iter: &TreeIter) -> TreeIter;
+
+    fn convert_path_to_child_path(&self, sorted_path: &TreePath) -> Option<TreePath>;
+
+    fn get_model(&self) -> Option<TreeModel>;
+
+    fn iter_is_valid(&self, iter: &mut TreeIter) -> bool;
+
+    fn reset_default_sort_func(&self);
+}
+
+impl<O: IsA<TreeModelSort>> TreeModelSortExt for O {
+    fn clear_cache(&self) {
+        unsafe {
+            ffi::gtk_tree_model_sort_clear_cache(self.to_glib_none().0);
+        }
+    }
+
+    fn convert_child_iter_to_iter(&self, child_iter: &TreeIter) -> Option<TreeIter> {
+        unsafe {
+            let mut sort_iter = TreeIter::uninitialized();
+            let ret = from_glib(ffi::gtk_tree_model_sort_convert_child_iter_to_iter(self.to_glib_none().0, sort_iter.to_glib_none_mut().0, mut_override(child_iter.to_glib_none().0)));
+            if ret { Some(sort_iter) } else { None }
+        }
+    }
+
+    fn convert_child_path_to_path(&self, child_path: &TreePath) -> Option<TreePath> {
+        unsafe {
+            from_glib_full(ffi::gtk_tree_model_sort_convert_child_path_to_path(self.to_glib_none().0, mut_override(child_path.to_glib_none().0)))
+        }
+    }
+
+    fn convert_iter_to_child_iter(&self, sorted_iter: &TreeIter) -> TreeIter {
+        unsafe {
+            let mut child_iter = TreeIter::uninitialized();
+            ffi::gtk_tree_model_sort_convert_iter_to_child_iter(self.to_glib_none().0, child_iter.to_glib_none_mut().0, mut_override(sorted_iter.to_glib_none().0));
+            child_iter
+        }
+    }
+
+    fn convert_path_to_child_path(&self, sorted_path: &TreePath) -> Option<TreePath> {
+        unsafe {
+            from_glib_full(ffi::gtk_tree_model_sort_convert_path_to_child_path(self.to_glib_none().0, mut_override(sorted_path.to_glib_none().0)))
+        }
+    }
+
+    fn get_model(&self) -> Option<TreeModel> {
+        unsafe {
+            from_glib_none(ffi::gtk_tree_model_sort_get_model(self.to_glib_none().0))
```

Comment:
This seems also need non-option return, as it created with non-option constructor.